### PR TITLE
Additional .NET examples

### PR DESCRIPTION
**Pull request by joephayes**
No GitHub issue.

Inspired by recent questions to support.

This adds two examples. One uses dotNetRDF and one uses Stardog's HTTP API. Both examples show how to list, drop and create databases.

Please note the Gradle build will continue to build/run the Trinity RDF example. I'll look into ways of making that more flexible.

Please add additional reviewers in case I missed anyone!

### CODE REVIEW

File: examples/dotnet/DotNetRDFConsoleSample/Program.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                //g.BaseUri = new Uri("http://example.org/graph");
+
+                //// write it to the store
+                //if (!stardogConn.IsReadOnly)
+                //{
+                //    stardogConn.SaveGraph(g);
+                //}
+
+                // Run a SPARQL query
+                object results = stardogConn.Query("SELECT * WHERE { GRAPH ?g { ?s ?p ?o } } LIMIT 100");
```
Comment (AlBaker) [inline examples/dotnet/DotNetRDFConsoleSample/Program.cs:75]: I guess dotNetRdf overloads query and has a generic response type, so you have to check the `is a`relationship on 76?

**Review by gcoluni — APPROVED** [review]
LGTM

File: examples/dotnet/HttpApiConsoleSample/Program.cs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+using System.IO;
+using System.IO.Compression;
+using System.Net;
+using System.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace HttpApiConsoleSample
+{
+    public class DatabaseList
```
Comment (AlBaker) [inline examples/dotnet/HttpApiConsoleSample/Program.cs:15]: Can you add comments here? I guess these are inner classes to use for serializing the REST calls?

File: examples/dotnet/HttpApiConsoleSample/Program.cs [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            const string STARDOG_PASSWORD = "admin";
+            const string DATABASE_NAME = "MyDB";
+
+            Uri baseServerUri = new Uri(SERVER_URL);
+            CredentialCache requestCredentialsCache = new CredentialCache();
+            requestCredentialsCache.Add(baseServerUri, "Basic", new NetworkCredential(STARDOG_USERNAME, STARDOG_PASSWORD));
+
+            // Get a list of all the server's databases
+
+            HttpWebRequest listDatabasesRequest = (HttpWebRequest)WebRequest.Create(new Uri(baseServerUri, "admin/databases"));
```
Comment (AlBaker) [inline examples/dotnet/HttpApiConsoleSample/Program.cs:44]: This is generic .NET `WebRequest` correct, and nothing specific to Stardog?  Do you think we should have these web service requests added to the dotNetRdf API, or perhaps as a dotNetRdfStardog package? dotNetRdf already has a stardog-specific portion, and we should make a determination if we want to bring it to parity w/ pystardog, stardog_rb, stardog.js and the like and round out the APIs.

File: examples/dotnet/HttpApiConsoleSample/Program.cs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                            }
+                        }
+
+                    }
+                }
+            }
+
+            // Create the database and add a triple from a file to it.
+
+            HttpWebRequest createDatabasesRequest = (HttpWebRequest)WebRequest.Create(new Uri(baseServerUri, "admin/databases"));
```
Comment (AlBaker) [inline examples/dotnet/HttpApiConsoleSample/Program.cs:85]: If we do leave this in here in the sample, and not having more extensions to dotNetRdf -- what do you think about moving this into its own class with a simplified method for executing it. So the "sample" class remains lightweight, and the "StardogAdmin" class encapsulates these REST calls.

**Review by AlBaker — APPROVED** [review]

**Review by evren — APPROVED** [review]
LGTM

File: examples/dotnet/DotNetRDFConsoleSample/Program.cs (hunk 1 above)
Comment (joephayes) [reply examples/dotnet/DotNetRDFConsoleSample/Program.cs:75]: Correct.

File: examples/dotnet/HttpApiConsoleSample/Program.cs (hunk 3 above)
Comment (joephayes) [reply examples/dotnet/HttpApiConsoleSample/Program.cs:44]: This is a generic .NET web request and they are already in the dotNetRDF (it uses similar calls to perform these functions). In the support call, the customer asked for an example of how to perform the create databases using the generic HttpWebRequest.

RE: feature parity with our other libraries, given all that we have added - virtual graph management, stored queries, etc. I think we should add the functionality to dotNetRDF. I think it makes the most sense to add it to the main dotNetRDF project or as some kind of separate plugin.

File: examples/dotnet/HttpApiConsoleSample/Program.cs (hunk 4 above)
Comment (joephayes) [reply examples/dotnet/HttpApiConsoleSample/Program.cs:85]: Yep - that makes perfect sense - FYI, I actually leaned heavily on the dotNetRDF call for how to properly configure the create databases (multipart request STINK!)- we could make a StardogAdmin library if we prefer something that is more lightweight then using the entire dotNetRDF package and we fully control it.

File: examples/dotnet/HttpApiConsoleSample/Program.cs (hunk 2 above)
Comment (joephayes) [reply examples/dotnet/HttpApiConsoleSample/Program.cs:15]: Done.

**Comment by joephayes** [comment]
@AlBaker - I addressed the comments, let me know if you're OK with me merging this PR

**Review by AlBaker — APPROVED** [review]

**Pull request merged**